mdf_forge/forge.py

```diff
@@ -437,6 +437,60 @@ def match_titles(self, titles):
         return self
 
 
+    def match_years(self, years=[], min = None, max = None, include=True):
+        """Add years and limits to the query.
+
+        Arguments:
+        years   (int, str or list of int or str): The years to match. Default [].
+        min     (int of str): The lower range of years to match..
+        max     (int of str): The upper range of years to match.
+        include (bool): If True, will add min and max years to the range. If False, they will be excluded.
+                          Default True.
+        limit (int): The maximum number of results to return.
+                     The max for this argument is the SEARCH_LIMIT imposed by Globus Search.
+        info (bool): If False, search will return a list of the results.
+                     If True, search will return a tuple containing the results list,
+                        and other information about the query. Default False.
+        Returns:
+        self (Forge): For chaining.
+        """
+        if not years and not min and not max:
+            print_("Year is not a valid input.")
+            return self
+
+        if years:
+            if not isinstance(years, list):
+                years = [years]
+            years_new = []
+            for year in years:
+                year = str(year)
+                if year.isdigit() and len(year) == 4:
+                    years_new.append(str(year))
+                else:
+                    print_("Year is not a valid input; use 'yyyy'.")
+                    return self
```

Comment:
It also makes more sense to check if a year is an integer by casting it with `int()` instead of checking if all the string characters are digits, especially because we don't really need to mandate four-digit years (despite the default for `min` being `None`, I foresee users using `0` as a minimum).

Reply:
All integers. Do we check for or enforce range? Any integer, negative?

Reply:
While it doesn't make sense for a user to enter negative years, or years in the future, as long as those values don't impact the functionality of the method I think they should be allowed. If a user wants to search for years between `-1` and `3000`, there's no real reason we shouldn't let them.

If negative years cause errors, though, they should be forbidden.

Reply:
OK, makes sense.

Reply:
No errors with negative integers. Done